internal/saucecloud/cloud.go

```diff
@@ -158,8 +162,12 @@ func (r *CloudRunner) collectResults(artifactCfg config.ArtifactDownload, result
 		r.logSuite(res)
 	}
 	close(done)
+	r.endTime = time.Now()
 
 	for _, rep := range r.Reporters {
+		if rpt, ok := rep.(*table.Reporter); ok {
```

Comment:
Code smell though. Neither the cloud nor the container module should care about the specific reporter. It'd be better if you could supply the total duration in some other shape or form.

How about this alternative approach:
What if the Job results were to contain the start/end time?
That way, the reporter can calculate the total runtime every time a new report is added (total = latest end time - oldest start time).

Reply:
ok! I'm also hesitated to add such type assertion here. Let me try your suggestion.